source/elements/oneMKL/source/domains/spblas/gemmoptimize.rst

```diff
@@ -0,0 +1,195 @@
+.. SPDX-FileCopyrightText: 2023 Intel Corporation
+..
+.. SPDX-License-Identifier: CC-BY-4.0
+
+.. _onemkl_sparse_optimize_gemm:
+
+optimize_gemm
+=============
+
+Performs internal optimizations for oneapi::mkl::sparse::gemm by analyzing
+the matrix structure.
+
+.. rubric:: Description and Assumptions
+
+The oneapi::mkl::sparse::optimize_gemm routine analyzes matrix structure
+and performs optimizations. Optimized data is then stored in
+the handle.
```

Comment:
lets explicitly add some language that discusses the two options here, and why either one exists (as well as talking about how the AB one should support calling with multiple B input size/shapes ...)

Reply:
Added a paragraph below.